apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
I was concerned about possible buffering. The write should be the equivalent of:

    write(fd, "CLIENT_RANDOM xxx yyy\n", ...);

and not

    write(fd, "CLIENT_RANDOM xxx yyy", ...);
    write(fd, "\n", 1);

Would this always result in a single write:

    BIO_puts(bio, line);
    BIO_write(bio, "\n", 1);
    BIO_flush(bio);